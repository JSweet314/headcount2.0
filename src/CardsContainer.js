import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';
import './styles/CardsContainer.css';

const CardsContainer = ({ districts, searchValue }) => {
  return (
    <div className='CardsContainer'> 
      {
        Object.keys(districts.stats)
          .map( (district, index) => {
            if (districts.stats[district].location.includes(searchValue)) {
              return <Card 
              district={districts.stats[district]}
              key={index} />
            }
            return null;      
          })
        }
      }
    </div>
  );
};

CardsContainer.propTypes = { 
  districts: PropTypes.object.isRequired
};

export default CardsContainer;