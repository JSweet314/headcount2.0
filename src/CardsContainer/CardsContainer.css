.CardsContainer {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  margin: auto;
  width: 85%;
}

@media screen and (max-width: 500px) {
  .CardsContainer {
    width: 100%;
  }
}
