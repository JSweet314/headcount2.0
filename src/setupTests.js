import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter() });

export const kinderData = [
  {
    "Location": "Colorado",
    "TimeFrame": 2007,
    "DataFormat": "Percent",
    "Data": 0.39465
  },
  {
    "Location": "Colorado",
    "TimeFrame": 2006,
    "DataFormat": "Percent",
    "Data": 0.33677
  },
  {
    "Location": "Colorado",
    "TimeFrame": 2005,
    "DataFormat": "Percent",
    "Data": 0.27807
  }
];