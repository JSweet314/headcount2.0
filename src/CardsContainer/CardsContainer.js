import React from 'react';
import Card from '../Card/Card.js';
import PropTypes from 'prop-types';
import './CardsContainer.css';

const CardsContainer = ({ districts, searchValue, selectLocation, selectedLocations }) => {

  return (
    <div className='CardsContainer'> 
      {
        districts.findAllMatches(searchValue)
          .map( (district, index) => {
              return <Card {...district}
                key={index} 
                selectLocation={selectLocation} 
                selectedLocations={selectedLocations} />;   
          })
      }
    </div>
  );
};

CardsContainer.propTypes = { 
  districts: PropTypes.object.isRequired,
  searchValue: PropTypes.string.isRequired,
  selectLocation: PropTypes.func.isRequired,
  selectedLocations: PropTypes.array.isRequired
};

export default CardsContainer;