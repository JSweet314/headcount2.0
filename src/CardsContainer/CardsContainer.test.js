import React from 'react';
import CardsContainer from './CardsContainer.js';
import DistrictRepository from '../helper.js';
import {kinderData} from '../setupTests.js';
import {shallow} from 'enzyme';

describe('CardsContainer', () => {
	const selectLocation = jest.fn();
	const selectedLocations = [];
	const districts = new DistrictRepository(kinderData);

	it('should have array of objects as props', () => {
		let cardsContainer = shallow(
      <CardsContainer 
        searchValue={''}
        districts={districts} 
        selectLocation={selectLocation} 
				selectedLocations={selectedLocations} />
    );
		expect(cardsContainer).toMatchSnapshot();
	});

})