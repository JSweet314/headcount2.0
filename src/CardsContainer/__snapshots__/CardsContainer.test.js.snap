// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsContainer should have array of objects as props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardsContainer
    districts={
        DistrictRepository {
            "stats": Object {
              "COLORADO": Object {
                "location": "COLORADO",
                "stats": Object {
                  "2005": 0.278,
                  "2006": 0.337,
                  "2007": 0.395,
                },
              },
            },
          }
    }
    searchValue=""
    selectLocation={[Function]}
    selectedLocations={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CompareContainer
          comparedAverage={undefined}
          searchValue=""
          selectLocation={[Function]}
          selectedLocations={Array []}
/>,
        Array [
          <Card
            location="COLORADO"
            selectLocation={[Function]}
            selectedLocations={Array []}
            stats={
                        Object {
                                    "2005": 0.278,
                                    "2006": 0.337,
                                    "2007": 0.395,
                                  }
            }
/>,
        ],
      ],
      "className": "CardsContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "comparedAverage": undefined,
          "searchValue": "",
          "selectLocation": [Function],
          "selectedLocations": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "location": "COLORADO",
          "selectLocation": [Function],
          "selectedLocations": Array [],
          "stats": Object {
            "2005": 0.278,
            "2006": 0.337,
            "2007": 0.395,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CompareContainer
            comparedAverage={undefined}
            searchValue=""
            selectLocation={[Function]}
            selectedLocations={Array []}
/>,
          Array [
            <Card
              location="COLORADO"
              selectLocation={[Function]}
              selectedLocations={Array []}
              stats={
                            Object {
                                          "2005": 0.278,
                                          "2006": 0.337,
                                          "2007": 0.395,
                                        }
              }
/>,
          ],
        ],
        "className": "CardsContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "comparedAverage": undefined,
            "searchValue": "",
            "selectLocation": [Function],
            "selectedLocations": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "location": "COLORADO",
            "selectLocation": [Function],
            "selectedLocations": Array [],
            "stats": Object {
              "2005": 0.278,
              "2006": 0.337,
              "2007": 0.395,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
